napari/settings/_tests/test_settings.py

```diff
@@ -300,6 +303,30 @@ def test_get_settings(tmp_path):
     assert str(s.config_path) == str(p)
 
 
+# test fallback loading from previous version
+def test_get_prev_ver_settings(tmp_path):
+    # prep a settings file for the previous version
+    data = "appearance:\n   theme: light"
+    prev_path = tmp_path / '0.4.16' / 'settings.yaml'
```

Comment:
this still will fail

Reply:
As I comment above. Napari thinks that is in version 0.1.0 and it is the downside of using `setuptools_scm` and `tox`.

Reply:
> It currently fails because the napari version is not monkeypatched in tests, so 0.4.17 is above the tested version `napari-0.1.dev1+g977c777`.
> 
> This is a bug in how tox works. It is hard to fix by comments as I cannot ad "Request change" for the import section of the test file.

Do we need to version check? the default path (config_path) is the current version path, so if that dir already exists, it's fine, we don't need to check vs. napari version. Just take the newest version and if it's current version it's fine.

Reply:
Yes. As it is possible that someone will have napari 0.4.16 and 0.4.18 and need to install and use napari 0.4.17. And importing settings from 0.4.18 may create a problem that you already observed and it cannot be solved even by calling reset.

Reply:
That case already will fail, because if 0.4.18 is the current version and present that will be used—that's `sources.append(config_path)`

Reply:
No. in my scenario, the current version will be 0.4.17

Reply:
OK, I think I have everything working correctly now.
The key was that the `config_path` was being set by `version_string` from `_appdirs` and not from `__version__`
Once I had the imports correct, I could use the pytest `monkeypatch`.
Now it works for me locally: no version dirs (uses napari parent), with 0.4.16 (uses that), with 0.4.17 (uses that), with 0.4.18 (doesn't use it, uses parent or 0.4.16 or 17 if present)

...and tests pass. 
🤞 for CI